Mobile friendly swap widget using Material UI's 'theme.spacing()'

secretshardul opened the pull request:
The current code uses hardcoded CSS dimensions which break responsiveness on mobile. They have been replaced by  `theme.spacing()` for adaptive resizing.

# Screenshots

## Mobile

![image](https://user-images.githubusercontent.com/49580849/121580503-712b8400-ca4a-11eb-9b74-17e9e5b2dd4a.png)

![image](https://user-images.githubusercontent.com/49580849/121580459-68d34900-ca4a-11eb-97a2-c58978e8678a.png)

## Desktop

![image](https://user-images.githubusercontent.com/49580849/121580298-345f8d00-ca4a-11eb-8b1c-4b0233a9b713.png)

![image](https://user-images.githubusercontent.com/49580849/121580563-7e487300-ca4a-11eb-971f-3e2f0c9db6a8.png)

secretshardul commented:
Travis linting error has been resolved.

armaniferrante commented:
Awesome. 

The new theme styling makes the component noticeably more cramped in the web ui. Is there a way to make these changes map more closely to the current component styling? For example, make it a little wider, taller, etc.

secretshardul commented:
> Awesome.
> 
> The new theme styling makes the component noticeably more cramped in the web ui. Is there a way to make these changes map more closely to the current component styling? For example, make it a little wider, taller, etc.

Took some creative liberty haha, so the design deviated a bit. Here's what I plan:
1. Taller: Add vertical padding
2. Wider: Add horizontal padding, increase width of `swapTokenSelectorContainer`.

I like the larger token icons, but should I revert that as well?

armaniferrante commented:
> > Awesome.
> > The new theme styling makes the component noticeably more cramped in the web ui. Is there a way to make these changes map more closely to the current component styling? For example, make it a little wider, taller, etc.
> 
> Took some creative liberty haha, so the design deviated a bit. Here's what I plan:
> 
> 1. Taller: Add vertical padding
> 2. Wider: Add horizontal padding, increase width of `swapTokenSelectorContainer`.
> 
> I like the larger token icons, but should I revert that as well?

I don't hate the new styling. Just want to make sure that if we change it, we do it with intention. Let's keep the styling the same for now, address the mobile responsiveness here, and we can consider changing sizes in a subsequent PR.

secretshardul commented:
Updated

> > > Awesome.
> > > The new theme styling makes the component noticeably more cramped in the web ui. Is there a way to make these changes map more closely to the current component styling? For example, make it a little wider, taller, etc.
> > 
> > 
> > Took some creative liberty haha, so the design deviated a bit. Here's what I plan:
> > 
> > 1. Taller: Add vertical padding
> > 2. Wider: Add horizontal padding, increase width of `swapTokenSelectorContainer`.
> > 
> > I like the larger token icons, but should I revert that as well?
> 
> I don't hate the new styling. Just want to make sure that if we change it, we do it with intention. Let's keep the styling the same for now, address the mobile responsiveness here, and we can consider changing sizes in a subsequent PR.

Updated to reflect original styling.

The original UI was not in the exact centre. You may notice a rightwards shift in the new UI. Here's what happens when width is increased in the old version.

https://github.com/project-serum/swap-ui/blob/1886a484cba208a64f6d30e43b17e6b7c8a73af4/example/src/App.tsx#L93

![Screenshot 2021-06-12 at 11 12 22 PM](https://user-images.githubusercontent.com/49580849/121785417-4de20f80-cbd7-11eb-91ab-82d46cef7365.png)

Fixed it in the same commit for it was breaking responsiveness on mobile screens.

armaniferrante commented:
Rebased on master.

armaniferrante commented:
The changes didn't quite restore the original UI dimensions. So I went ahead and used your original spacing parameters.
